futures-util/src/async_await/mod.rs

```diff
@@ -3,37 +3,37 @@
 //! This module contains a number of functions and combinators for working
 //! with `async`/`await` code.
 
-use futures_core::future::Future;
-use futures_core::stream::Stream;
-
-#[doc(hidden)]
-pub use futures_core::future::FusedFuture;
-#[doc(hidden)]
-pub use futures_core::stream::FusedStream;
+use futures_core::future::{Future, FusedFuture};
+use futures_core::stream::{Stream, FusedStream};
 
 #[macro_use]
 mod poll;
+#[allow(unreachable_pub)] // https://github.com/rust-lang/rust/issues/57411
```

Comment:
I don't think this is that issue, that seems to specifically be a bad interaction between a fix-suggestion and having a deny-lint active (deleting the `pub use` in the example there results in a working program).

It seems since we don't make `futures-util::async_await` a `pub mod` and instead `pub use async_await::*;` these _can_ all be `pub(crate)` and still work.

Reply:
Oh, no, I didn't realise re-exports could further constrain the visibility of the re-exported item, rather than just having a visibility applied to the re-export itself. I think this is more likely https://github.com/rust-lang/rust/issues/64762

Reply:
Thanks, it certainly seems more likely to be https://github.com/rust-lang/rust/issues/64762. 
(Actually, the code mentioned in [this comment](https://github.com/rust-lang/rust/issues/64762#issuecomment-567555428) looks almost the same as in this PR's case, except for the module itself warned.)